### ISSUE TITLE
add do_smallscale feature

### ISSUE THREAD

**Issue by JOThurgood** [AMReX-Astro/MAESTROeX#52]
It might be useful to add a `do_smallscale` feature and associated test problems that is analogous to the feature in the old MAESTRO code. 

(Namely, setting rho_0 = rhoh_0 = 0,  not evolving the background, and using div U = S constraint.)

Since it seems like an approachable first issue, I'll attempt this and then issue a pull request once it is ready.

**Comment by JOThurgood** [comment]
Turns out  the `do_smallscale` stuff is already in the solver. 

The parameter wasn't available to the fortran routines, and there was a misleading comment in `MaestroAdvance.cpp`, that led me to think otherwise. It was also possible to inadvertently set `do_smallscale = T` without the corresponding `evolve_base_state=F` and `beta0_type=3`. 

I've prepared a small commit that makes it available to fortran modules, and has a runtime check for the consistency of those parameters. If inconsistent it returns a warning and aborts. I will submit a PR later today.
